### ISSUE TITLE
Should we switch to `emacsmirror/nongnu_elpa` rather than `Savannah nongnu_elpa`?

### ISSUE THREAD

**Issue by milanglacier** [raxod502/straight.el#1189]
Savannah has recently experienced accessibility issues frequently, possibly due to frequent DDoS attacks. Consequently, we are unable to update packages or bootstrap Emacs.  

Should we consider switching from `https://git.savannah.gnu.org/git/emacs/nongnu.git` to `https://github.com/emacsmirror/nongnu_elpa.git` in the `straight-use-recipes` call when defining the `nongnu-elpa` recipe in `bootstrap.el`?

**Comment by raxod502** [comment]
That seems reasonable, yes, if there's reasonable data parity between the two. For backwards compatibility and customizability, I might suggest introducing a new recipe repository name and user option, same as for GNU ELPA:

https://github.com/radian-software/straight.el/blob/b89c950b00470dac0299eb0a731d5c72c296133c/bootstrap.el#L124-L132

**Comment by martenlienen** [comment]
Would it be an acceptable first solution to fix the typo in #1147 and merge that? Non-GNU ELPA has had a lot of problems in the past two weeks and then users could switch out the repo URL for `https://github.com/emacsmirror/nongnu_elpa` or some other mirror. Alternatively, accessing the repo over its git URL `git://git.savannah.gnu.org/emacs/nongnu.git` also seemed to work fine, compared to the http that is in the code right now.

**Comment by raxod502** [comment]
Sure. As a local workaround, you can also use `straight-recipe-overrides` in your init-file before loading `straight.el`, which lets you override the recipe directly, no upstream changes needed.

**Comment by martenlienen** [comment]
The following snippet before the straight.el bootstrap section in my `init.el` works for me.
```elisp
(custom-set-variables
 '(straight-recipe-overrides '((nil
                                (nongnu-elpa :type git
                                             :repo "https://github.com/emacsmirror/nongnu_elpa"
                                             :depth (full single-branch)
                                             :local-repo "nongnu-elpa"
                                             :build nil)
                                (ws-butler :type git
                                           :repo "https://github.com/emacsmirror/nongnu_elpa"
                                           :branch "elpa/ws-butler"
                                           :depth (full single-branch)
                                           :local-repo "ws-butler")))))
```
ws-butler is the package that triggered this for me in the first place when it moved to NonGNU ELPA.

**Comment by SeanMooney** [comment]
> The following snippet before the straight.el bootstrap section in my `init.el` works for me.
> 
> (custom-set-variables
>  '(straight-recipe-overrides '((nil
>                                 (nongnu-elpa :type git
>                                              :repo "https://github.com/emacsmirror/nongnu_elpa"
>                                              :depth (full single-branch)
>                                              :local-repo "nongnu-elpa"
>                                              :build nil)
>                                 (ws-butler :type git
>                                            :repo "https://github.com/emacsmirror/nongnu_elpa"
>                                            :branch "elpa/ws-butler"
>                                            :depth (full single-branch)
>                                            :local-repo "ws-butler")))))
> 
> ws-butler is the package that triggered this for me in the first place when it moved to NonGNU ELPA.

This works well for me too, thanks for the suggestion.

**Comment by milanglacier** [comment]
An alternative approach, particularly if you feel you are concerning about "overriding" something in your config.

```lisp
(setq straight-recipe-repositories '(org-elpa melpa gnu-elpa-mirror el-get emacsmirror-mirror))

(straight-use-package
 '(ws-butler :type git
             :repo "https://github.com/emacsmirror/nongnu_elpa"
             :branch "elpa/ws-butler"
             :depth (full single-branch)
             :local-repo "ws-butler"))
```

Given that I haven't encountered packages only available on NonGNU-ELPA and absent from MELPA, directly removing NonGNU-ELPA from the package archives might not be a big probelm.

 To use `ws-butler`, one can simply retrieve it directly from the GitHub mirror.

**Comment by raxod502** [comment]
Per https://github.com/radian-software/straight.el/pull/1147, the URL for NonGNU ELPA can now be customized. Furthermore per #1198 you can now customize the set of default recipe repositories.

**Comment by raxod502** [comment]
https://github.com/radian-software/straight.el/pull/1215 implements the change suggested here.

See also https://github.com/radian-software/straight.el/issues/1192, https://github.com/radian-software/straight.el/issues/1191.
